rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -98,7 +99,21 @@ class Publisher
       // *INDENT-ON*
     }
     if (store_intra_process_message_) {
-      uint64_t message_seq = store_intra_process_message_(intra_process_publisher_id_, msg);
+      // Take the pointer from the unique_msg, release it and pass as a void *
```

Comment:
This seemed weird to me at first but it makes sense that it's necessary to transfer the ownership of the unique_ptr.